lib/recursive-methods.rb

```diff
@@ -1,49 +1,100 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+    if n < 0 
+        raise ArgumentError.new("Number must be greater than zero!")
+    elsif n == 1
+        return n
+    elsif n == 0
+        return 1
+    else
+        return n * factorial(n - 1)
+    end
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(n)
 def reverse(s)
```

Comment:
Since you are creating a new array with `s[1..-1]` in each recursive call this is actually O(n^2)